protocol/daemons/slinky/client/sidecar_version_checker.go

```diff
@@ -0,0 +1,72 @@
+package client
+
+import (
+	"context"
+	"fmt"
+
+	"cosmossdk.io/log"
+	"github.com/hashicorp/go-version"
+
+	oracleclient "github.com/skip-mev/connect/v2/service/clients/oracle"
+	"github.com/skip-mev/connect/v2/service/servers/oracle/types"
+)
+
+// SidecarVersionChecker is a lightweight process run in a goroutine by the slinky client.
+// Its purpose is to query the running sidecar version and check if it is at least a minimum
+// acceptable version.
+type SidecarVersionChecker interface {
+	Start(ctx context.Context) error
+	Stop()
+	CheckSidecarVersion(context.Context) error
+}
+
+// SidecarVersionCheckerImpl implements the SidecarVersionChecker interface.
+type SidecarVersionCheckerImpl struct {
+	slinky oracleclient.OracleClient
+	logger log.Logger
+}
+
+func NewSidecarVersionChecker(slinky oracleclient.OracleClient, logger log.Logger) SidecarVersionChecker {
+	return &SidecarVersionCheckerImpl{
+		slinky: slinky,
+		logger: logger,
+	}
+}
+
+// Start initializes the underlying connections of the SidecarVersionChecker.
+func (s *SidecarVersionCheckerImpl) Start(
+	ctx context.Context) error {
+	return s.slinky.Start(ctx)
+}
+
+// Stop closes all existing connections.
+func (s *SidecarVersionCheckerImpl) Stop() {
+	_ = s.slinky.Stop()
+}
+
+func (p *SidecarVersionCheckerImpl) CheckSidecarVersion(ctx context.Context) error {
```

Comment:
nit: Same as the AI comment, name the variable `s`? Assume this was copied from the price checker (which is called `p`)?